src/Microsoft.ML.TorchSharp/NasBert/NerTrainer.cs

```diff
@@ -334,6 +370,41 @@ private protected override Delegate CreateGetter(DataViewRow input, int iinfo, T
 
             }
 
+            private void CondenseOutput(ref VBuffer<UInt32> dst, string sentence, Tokenizer tokenizer, TensorCacher outputCacher)
+            {
+                var pre = tokenizer.PreTokenizer.PreTokenize(sentence);
+                TokenizerResult encoding = tokenizer.Encode(sentence);
```

Comment:
Does this tokenize the same sentence that was previously tokenized as input to the model?

Reply:
It does. I didn't change it to use what was already encoded cause of what I would have to change (base classes and other classes that use the base class) and that this isn't time consuming at all compared to the actual processing of the input. I am happy to make the changes though.

Reply:
How does the implementation we ported this from handle the same scenario?  Does it recalculate the tokenization multiple times or does it somehow flow it through?
We could always keep this the way it is, and come back to fix it if the perf matters, though I was wondering if we somehow flowed more information about the tokenization through the data we might be able to avoid having to expose new API on the tokenizer.